error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[1]’ may be used uninitialized in this function

I am seeing the following error when compiling cub
```
[ 91%] Built target cub.cpp14.test.device_scan_by_key.types_1
/home/gaoxiang/cub/test/test_device_reduce.cu: In function ‘void SolveAndTest(HostInputIteratorT, DeviceInputIteratorT, OffsetT, OffsetT, BeginOffsetIteratorT, EndOffsetIteratorT, BeginOffsetIteratorT, EndOffsetIteratorT, ReductionOpT) [with Backend BACKEND = CUB; OutputValueT = char; HostInputIteratorT = char*; DeviceInputIteratorT = char*; OffsetT = int; BeginOffsetIteratorT = int*; EndOffsetIteratorT = int*; ReductionOpT = cub::Sum]’:
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[0]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  871 |     DiscardOutputIterator<OffsetT> discard_itr;
      |                                       ^~~~~~~~~  
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[1]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[2]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu: In function ‘void SolveAndTest(HostInputIteratorT, DeviceInputIteratorT, OffsetT, OffsetT, BeginOffsetIteratorT, EndOffsetIteratorT, BeginOffsetIteratorT, EndOffsetIteratorT, ReductionOpT) [with Backend BACKEND = CUB; OutputValueT = unsigned char; HostInputIteratorT = unsigned char*; DeviceInputIteratorT = unsigned char*; OffsetT = int; BeginOffsetIteratorT = int*; EndOffsetIteratorT = int*; ReductionOpT = cub::Sum]’:
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[0]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[1]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[2]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu: In function ‘void SolveAndTest(HostInputIteratorT, DeviceInputIteratorT, OffsetT, OffsetT, BeginOffsetIteratorT, EndOffsetIteratorT, BeginOffsetIteratorT, EndOffsetIteratorT, ReductionOpT) [with Backend BACKEND = CUB; OutputValueT = int; HostInputIteratorT = char*; DeviceInputIteratorT = char*; OffsetT = int; BeginOffsetIteratorT = int*; EndOffsetIteratorT = int*; ReductionOpT = cub::Sum]’:
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[0]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[1]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[2]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu: In function ‘void TestByOp(HostInputIteratorT, DeviceInputIteratorT, OffsetT, OffsetT, BeginOffsetIteratorT, EndOffsetIteratorT, BeginOffsetIteratorT, EndOffsetIteratorT) [with Backend BACKEND = CUB; OutputT = char; HostInputIteratorT = char*; DeviceInputIteratorT = char*; OffsetT = int; BeginOffsetIteratorT = int*; EndOffsetIteratorT = int*]’:
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[0]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[1]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[2]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[0]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[1]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[2]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[0]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[1]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/gaoxiang/cub/test/test_device_reduce.cu:871:39: error: ‘discard_itr.cub::DiscardOutputIterator<int>::pad[2]’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
```
The following version of compiler are used:
```
$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Thu_Nov_18_09:45:30_PST_2021
Cuda compilation tools, release 11.5, V11.5.119
Build cuda_11.5.r11.5/compiler.30672275_0

$ gcc-10 --version
gcc-10 (Arch Linux 10.3.0-2) 10.3.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/cub/iterator/discard_output_iterator.cuh
+++ b/cub/iterator/discard_output_iterator.cuh
@@ -87,7 +87,7 @@ private:
 
 #if defined(_WIN32) || !defined(_WIN64)
     // Workaround for win32 parameter-passing bug (ulonglong2 argmin DeviceReduce)
-    OffsetT pad[CUB_MAX(1, (16 / sizeof(OffsetT) - 1))];
+    OffsetT pad[CUB_MAX(1, (16 / sizeof(OffsetT) - 1))] = {};
 #endif
 
 public:

